app/src/main/java/de/westnordost/streetcomplete/quests/cycleway/AddCycleway.kt

```diff
@@ -194,54 +193,55 @@ class AddCycleway(private val countryInfos: CountryInfos) : OsmElementQuestType<
         val cyclewayKey = "cycleway:" + side.value
         when (cycleway) {
             NONE, NONE_NO_ONEWAY -> {
-                changes.addOrModify(cyclewayKey, "no")
+                tags[cyclewayKey] = "no"
             }
             EXCLUSIVE_LANE, ADVISORY_LANE, UNSPECIFIED_LANE -> {
-                changes.addOrModify(cyclewayKey, "lane")
+                tags[cyclewayKey] = "lane"
                 if (directionValue != null) {
-                    changes.addOrModify("$cyclewayKey:oneway", directionValue)
+                    tags["$cyclewayKey:oneway"] = directionValue
+                }
+                if (cycleway == EXCLUSIVE_LANE) {
+                    tags["$cyclewayKey:lane"] = "exclusive"
+                } else if (cycleway == ADVISORY_LANE) {
+                    tags["$cyclewayKey:lane"] = "advisory"
                 }
-                if (cycleway == EXCLUSIVE_LANE)
-                    changes.addOrModify("$cyclewayKey:lane", "exclusive")
-                else if (cycleway == ADVISORY_LANE)
-                    changes.addOrModify("$cyclewayKey:lane","advisory")
             }
             TRACK -> {
-                changes.addOrModify(cyclewayKey, "track")
+                tags[cyclewayKey] = "track"
                 if (directionValue != null) {
-                    changes.addOrModify("$cyclewayKey:oneway", directionValue)
+                    tags["$cyclewayKey:oneway"] = directionValue
                 }
-                if (changes.getPreviousValue("$cyclewayKey:segregated") == "no") {
-                    changes.modify("$cyclewayKey:segregated", "yes")
+                if (tags.containsKey("$cyclewayKey:segregated")) {
+                    tags["$cyclewayKey:segregated"] = "yes"
```

Comment:
Yes, the tag should be changed to what the user answered, even if the key was previously a trash tag value. Though, I primarily changed this because I merged `modify`, `addOrModify` and `add` into one function (`operator fun set`) and it is OK if the statement in line 215 is executed when cyclewayKey:segregated is already "yes". It then simply does nothing.